webapp/channels/src/components/advanced_text_editor/scheduled_post_indicator/scheduled_post_indicator.tsx

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+import {type match} from 'react-router-dom';
+import {NavLink, useRouteMatch} from 'react-router-dom';
+
+import {showChannelOrThreadScheduledPostIndicator} from 'mattermost-redux/selectors/entities/scheduled_posts';
+
+import {SCHEDULED_POST_TIME_RANGES, scheduledPostTimeFormat} from 'components/drafts/panel/panel_header';
+import Timestamp from 'components/timestamp';
+
+import {Locations} from 'utils/constants';
+
+import type {GlobalState} from 'types/store';
+
+import './scheduled_post_indicator.scss';
+
+type Props = {
+    location: string;
+    channelId: string;
+    postId: string;
+}
+
+export default function ScheduledPostIndicator({location, channelId, postId}: Props) {
+    // we use RHS_COMMENT for RHS and threads view, and CENTER for center channel.
+    // get scheduled posts of a thread if in RHS or threads view,
+    // else, get those for the channel.
+    // Fetch scheduled posts for the thread when opening a thread, and fetch for channel
+    // when opening from center channel.
+    const id = location === Locations.RHS_COMMENT ? postId : channelId;
+    const scheduledPostData = useSelector((state: GlobalState) => showChannelOrThreadScheduledPostIndicator(state, id));
+    const match: match<{team: string}> = useRouteMatch();
+
```

Comment:
This would cause the following `useMemo` to be called conditionally, which is an error in React. If I move this below the useMemo, its effectively the same code. So, I just move the "return if null" case above to resolve this.